packages/node-patches/src/fs.ts

```diff
@@ -0,0 +1,345 @@
+import { Stats, Dirent, Dir } from 'fs';
+import * as path from 'path';
+import * as util from 'util';
+// using require here on purpose so we can override methods with any
+// also even though imports are mutable in typescript the cognitive dissonance is too high because es modules
+const _fs = require('fs');
+
+//tslint:disable-next-line:no-any
+export const patcher = (fs: any, root: string) => {
+  fs = fs || _fs;
+  root = root || process.env.BAZEL_SYMLINK_PATCHER_ROOT || '';
+  if (root) root = fs.realpathSync(root);
```

Comment:
I think I have two high level concerns about `root`:

1. Would it be cleaner to use `null` or `false` instead of the empty string to mark "no root provided"?
2. At least from the first few patches, it looks like the patcher doesn't have anything meaningful to do without `root`. If that's true: Could `root` be required instead of handling its absence?

Reply:
this is a general thing and where ts is changing my habits.  Its more difficult to deal with variables that are multiple types in typescript i'd need to add some typeof guard so i would have to concat strings to it every time i used it. In this case i agree that it should be required so `if(!root) return;` would fix any concerns.